src/drivers/s3.ts

```diff
@@ -36,7 +48,7 @@ export interface S3DriverOptions {
   /**
    * The name of the bucket.
    */
-  bucket: string;
+  bucket?: string;
```

Comment:
`endpoint` should not be deprecated because of a new feature. If you like to introduce `sessionToken` please open one PR for it.

If you want to introduce `base`, another PR.